src/paymentservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/paymentservice/CrcController.java

```diff
@@ -0,0 +1,49 @@
+package ca.bc.gov.open.pssg.rsbc.dps.paymentservice;
+
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.configuration.CrcProperties;
+import ca.bc.gov.open.pssg.rsbc.dps.paymentservice.types.BeanstreamEndpointResponse;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@EnableConfigurationProperties(CrcProperties.class)
+public class CrcController {
+
+    private static final Logger logger = LogManager.getLogger(CrcController.class);
+    private final CrcProperties crcProperties;
+
+    public CrcController(CrcProperties crcProperties) {
+        this.crcProperties = crcProperties;
+    }
+
+
+    @RequestMapping(value = "/getBeanstreamEndpoints",
+            produces = { "application/xml" },
+            method = RequestMethod.GET)
+    @ApiOperation(value = "Generates application callback endpoints for Payment Service", notes = "", response = BeanstreamEndpointResponse.class, tags={ "PaymentServices"})
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successful operation", response = BeanstreamEndpointResponse.class) })
+    public BeanstreamEndpointResponse getBeanstreamEndpoints() {
+
+        if (StringUtils.isEmpty(crcProperties.getApproved()) || StringUtils.isEmpty(crcProperties.getDeclined()) || StringUtils.isEmpty(crcProperties.getError())) {
+            logger.fatal("CRC endpoints are not set properly in the config. Failed to read the properties. ");
```

Comment:
do we need a stack trade here as well with the fatal error?

Reply:
yeah I don't think it is a fatal error too...

Reply:
and there is no exception here...